apps/nextjs-app-router/src/middleware.ts

```diff
@@ -0,0 +1,33 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { match } from '@formatjs/intl-localematcher'
+import Negotiator from 'negotiator'
+import { defaultLocale, locales } from '@/localization'
+
+function getLocale(request: NextRequest) {
+  const acceptLanguageHeader = request.headers.get('accept-language')
+  let languages = new Negotiator({
+    headers: { 'accept-language': acceptLanguageHeader ?? undefined },
```

Comment:
Why do we need to coalesce `undefined` to `undefined`? Can't we do

```ts
new Negotiator({ 'Accept-Language': request.headers.get('Accept-Language') }).languages()
```

Or even cleaner

```ts
new Negotiator(request.headers).languages()
```

?